test/unit/org/apache/cassandra/auth/MutualTlsInternodeAuthenticatorTest.java

```diff
@@ -0,0 +1,163 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.auth;
+
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import org.apache.cassandra.SchemaLoader;
+import org.apache.cassandra.config.Config;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.exceptions.ConfigurationException;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.service.StorageService;
+
+import static org.apache.cassandra.auth.AuthTestUtils.loadCertificateChain;
+import static org.apache.cassandra.auth.IInternodeAuthenticator.InternodeConnectionDirection.INBOUND;
+import static org.apache.cassandra.auth.IInternodeAuthenticator.InternodeConnectionDirection.OUTBOUND;
+import static org.apache.cassandra.config.CassandraRelevantProperties.CASSANDRA_CONFIG;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(Parameterized.class)
+public class MutualTlsInternodeAuthenticatorTest
+{
+    private static final String VALIDATOR_CLASS_NAME = "validator_class_name";
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+    @Parameterized.Parameter(0)
+    public String certificatePath;
+    @Parameterized.Parameter(1)
+    public String identity;
+
+    @Parameterized.Parameters()
+    public static Collection<Object[]> versions()
+    {
+        return Collections.singletonList(new Object[]{ "auth/SampleMtlsClientCertificate.pem", "spiffe://testdomain.com/testIdentifier/testValue" });
+    }
+
+    @BeforeClass
+    public static void initialize()
+    {
+        CASSANDRA_CONFIG.setString("cassandra-mtls.yaml");
+        SchemaLoader.loadSchema();
+        DatabaseDescriptor.daemonInitialization();
+        StorageService.instance.initServer(0);
+    }
+
+    @Before
+    public void before()
+    {
+        Config config = DatabaseDescriptor.getRawConfig();
+        config.server_encryption_options = config.server_encryption_options.withOutboundKeystore("test/conf/cassandra_ssl_test_outbound.keystore")
+                                                                           .withOutboundKeystorePassword("cassandra");
```

Comment:
Do these test stores have any expiry times? If so how would somebody regenerate them to fix tests.

Reply:
@jonmeredith  Yes the certificate does have an expiry. However these certificates are not used for SSL/TLS handshake.  They are used only for testing extracting identities and in authenticators, so even if they expire they do not make the tests to fail.